tests/ClientProvider/TestSetupHelper.php

```diff
@@ -56,8 +56,12 @@ public function setUpForProtectedDevBranch(
             $branchName = $clientProvider->getDevBranchName();
             // dev can create & delete branches in production
             $devBranches = new DevBranches($clientProvider->getDefaultClient(['token' => STORAGE_API_DEVELOPER_TOKEN]));
-            $this->deleteBranchesByPrefix($devBranches, $branchName);
-            $branch = $devBranches->createBranch($branchName);
+
+            $branch = $this->getExistingBranches($devBranches, $branchName);
+
+            if ($branch === null) {
+                $branch = $devBranches->createBranch($branchName);
```

Comment:
vytvaram novu branch len ked neexistuje pre dany test, vsade kde sa to pouziva sa nejakou formou premazavaju buckty/metadata a pod. takze test prechadza a myslim si e to pomaha na err You don't have access to resource.